sycl/test-e2e/LLVMIntrinsicLowering/bitreverse.cpp

```diff
@@ -0,0 +1,213 @@
+// Test that llvm.bitreverse is lowered correctly by llvm-spirv
+
+// UNSUPPORTED: hip || cuda
+
+// RUN: %{build} -o %t.out -O2
```

Comment:
@mdtoguchi, I have a question about the driver. @LU-JOHN and I discussed the ways to validate that LLVM IR driver passes to llvm-spirv tool has `llvm.bitreverse` intrinsic. Do you know if we have an option to either dump LLVM IR we pass to llvm-spirv tool or maybe we can pass additional command line options to `llvm-spirv` tool (which will ask the tool to dump the input).
I know that `-save-temps` potentially could do that, but it seems like an overkill. Also I don't know if it will play nicely with CI system. It produces a lot of files we don't need for our task, which CI might not clean-up.

Reply:
There is a new option `-fsycl-device-code-dump=<dir>` you can use to dump the device code that is going to be passed to `llvm-spirv`.

Another option is to use `-Xspirv-translator <opt>` to pass an `<opt>` to the translator step.

Reply:
Great! Thanks a lot! @LU-JOHN, let's try using `-fsycl-device-code-dump=<dir>` to verify we LLVM IR we pass to the `llvm-spirv` tool has bitreverse intrinsic.

Reply:
Is the option -fsycl-device-code-dump available in SYCLOS now?  I couldn't get it to work:

```
clang++ -fsycl bitreverse.cpp -O2 -w -fsycl-device-code-dump=.
clang++: error: unknown argument: '-fsycl-device-code-dump=.'
```

Reply:
My bad - syntax of the option is `-fsycl-dump-device-code=<dir>`

Reply:
> There is a new option -fsycl-device-code-dump=<dir> you can use to dump the device code that is going to be passed to llvm-spirv

I got the option to work, but I see a SPV file generated by llvm-spirv, not the BC input for llvm-spirv.

Reply:
Actually, the SPV file is sufficient.  I can use it to verify that the lowered names appear in the SPV.

Reply:
I've updated the testcase to ensure that the SPV dump has the expected lowerings.